packages/devtools_app/lib/src/screens/logging/logging_screen_v2/logging_model.dart

```diff
@@ -0,0 +1,117 @@
+// Copyright 2024 The Chromium Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+import 'dart:async';
+
+import 'package:flutter/foundation.dart';
+
+import '../../../shared/utils.dart';
+import 'logging_controller_v2.dart';
+import 'logging_table_row.dart';
+import 'logging_table_v2.dart';
+
+/// A class for holding state and state changes relevant to [LoggingControllerV2]
+/// and [LoggingTableV2].
+///
+/// The [LoggingTableV2] table uses variable height rows. This model caches the
+/// relevant heights and offsets so that the row heights only need to be calculated
+/// once per parent width.
+class LoggingTableModel extends ChangeNotifier {
+  LoggingTableModel() {
+    _worker = InterruptableChunkWorker(
+      callback: getFilteredLogHeight,
+      progressCallback: (progress) => _cacheLoadProgress.value = progress,
+    );
+  }
+
+  final _logs = <LogDataV2>[];
+  final _filteredLogs = <LogDataV2>[];
+  final _selectedLogs = <int>{};
+
+  final cachedHeights = <int, double>{};
+  final cachedOffets = <int, double>{};
+
+  late final InterruptableChunkWorker _worker;
+
+  /// Represents the state of reloading the height caches.
+  ///
+  /// When null, then the cache is not loading.
+  /// When double, then the value is represents how much progress has been made.
+  ValueListenable<double?> get cacheLoadProgress => _cacheLoadProgress;
+  final _cacheLoadProgress = ValueNotifier<double?>(null);
+
+  @override
+  void dispose() {
+    super.dispose();
+    _cacheLoadProgress.dispose();
+  }
+
+  /// Update the width of the table.
+  ///
+  /// This will flush all of the calculated heights, and recalculate their heights
+  /// in the background.
+  set tableWidth(double width) {
+    _tableWidth = width;
+    cachedHeights.clear();
+    cachedOffets.clear();
+    unawaited(_preFetchRowHeights());
+  }
+
+  /// Get the filtered log at [index].
+  LogDataV2 getFilteredLog(int index) => _filteredLogs[index];
+
+  double _tableWidth = 0.0;
+
+  /// The total number of logs being held by the [LoggingTableModel].
+  int get logCount => _logs.length;
+
+  /// The number of filtered logs.
+  int get filteredLogCount => _filteredLogs.length;
+
+  /// The number of selected logs.
+  int get selectedLogCount => _selectedLogs.length;
```

Comment:
how can there be more than one selection? Are you referring to text selection here?

Reply:
In a future iteration, the ability to select multiple logs will be added. With checkboxes on the left side likely.
Otherwise the user will be limited to actions that are just in the current view. Text selection on scrolling views means that some of the widgets disapears and cause buggy behaviour.

Reply:
We should raise the priority of text selection issues like this with the framework team. @Piinks @goderbauer do you know if text selection in scrolling views is something that is actively being worked on?

Reply:
I think it is, is there an open issue you can point us to?

Reply:
I've posted a new issue for a different selection bug though.
https://github.com/flutter/devtools/issues/7893